clients/cobol-lsp-vscode-extension/src/services/reporter/TelemetryReporterImpl.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+import * as fs from "fs";
+import * as path from "path";
+import * as vscode from "vscode";
+import TelemetryReporter from "vscode-extension-telemetry";
+import {EXTENSION_ID, TELEMETRY_DEFAULT_CONTENT} from "../../constants";
+import {ExtensionUtils} from "../settings/util/ExtensionUtils";
+import {TelemetryEvent} from "./model/TelemetryEvent";
+import {TelemetryMeasurement} from "./model/TelemetryMeasurement";
+import {TelemetryReport} from "./TelemetryReport";
+
+export class TelemetryReporterImpl implements TelemetryReport {
+    public static getInstance(): TelemetryReporterImpl {
+        if (!TelemetryReporterImpl.instance) {
+            TelemetryReporterImpl.instance = new TelemetryReporterImpl(this.getTelemetryKeyId());
+        }
+        return TelemetryReporterImpl.instance;
+    }
```

Comment:
Why do you need that lazy initialization?

Reply:
Because ```this.getTelemetryKeyId()``` will scan the filesystem to read the content of the file and since its value is not supposed to change I prefer to have only one instance of the ```TelemetryReporterImpl```